src/uidl-definitions/schemas/new-component.json

```diff
@@ -0,0 +1,279 @@
+              "required": ["type", "defaultValue"],
+              "properties": {
+                "type": {"type":"string", "enum": ["string", "boolean", "number", "object", "func", "array", "router"]},
+                "values": {
+                  "type": "array",
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "value": {},
+                      "meta": { 
+                        "type": "object",
+                        "properties": {
+                          "componentName": { "type": "string" },
+                          "path": { "type": "string" },
+                          "fileName": { "type": "string" }
+                        }
+                      },
+                      "transitions": { "type": "object" }
+                    }
+                  }
+                },
+                "defaultValue": {},
+                "actions": {"type": "array"}
+              }
+            }
+          }
+        },
+        "style": {},
+        "node": {
+          "oneOf": [{
+            "$ref": "#/definitions/staticValue" 
+          },{
+            "$ref": "#/definitions/dynamicReference"
+          },{
+            "$ref": "#/definitions/repeatNode" 
+          },{
+           "$ref": "#/definitions/elementNode"
+          },{
+            "$ref": "#/definitions/conditionalNode"
+          }]
+        },
+        "staticValue": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": [ "type", "content" ],
+          "properties": {
+            "type": {
+              "type": "string",
+              "enum":  ["static"]
+              },
+            "content": { 
+              "oneOf": [ 
+                { "type": "string" },
+                { "type": "number" },
+                { "type": "boolean" },
+                { "type": "array"} 
+              ] 
+            }
+          }
+        },
+        "dynamicReference": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": [ "type", "content" ],
+          "properties": {
+            "type": {
+              "type": "string",
+              "enum":  ["dynamic"]
+            },
+            "content": { 
+              "required": [ "referenceType", "id" ],
+              "additionalProperties": false,
+              "referenceType" :{
+                "type": "string",
+                "enum":  ["prop", "state", "local", "attr"]
+              },
+              "id": { "type": "string" }
+            }
+          }
+        },
+        "repeatNode": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": [ "type", "content" ],
+          "properties": {
+            "type": {
+              "type": "string",
+              "enum":  ["repeat"]
+            },
+            "content": {
+              "$ref": "#/definitions/repeatContent"
```

Comment:
no need to split this if it's not reused somewhere else

Reply:
I'd argue we could split it for ease of editing. I'd rather have the content of the repeat in a separate file for example and only focus on it if we make changes in the future.